test/money-market/bTokens.test.ts

```diff
@@ -0,0 +1,146 @@
+import { ethers } from 'hardhat';
+import { deployBTokens } from '../helpers/money-market';
+import { fork } from '../helpers';
+import { ADDRESS } from '../../constant';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { BErc20Delegator, Comptroller, ERC20 } from '../../typechain-types';
+import { faucetToken } from '../helpers/paraswap';
+import { BigNumber, utils } from 'ethers';
+import { expect } from 'chai';
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+describe('Money Market', async () => {
+  let admin: SignerWithAddress;
+  let alice: SignerWithAddress;
+  let comptroller: Comptroller;
+
+  let bWETH: BErc20Delegator;
+  let bBAL: BErc20Delegator;
+  let bUSDC: BErc20Delegator;
+  let bWBTC: BErc20Delegator;
+  let bOHM: BErc20Delegator;
+
+  before(async () => {
+    await fork(1);
+    [admin, alice] = await ethers.getSigners();
+
+    const market = await deployBTokens(admin.address);
+
+    comptroller = market.comptroller;
+    bWETH = market.bWETH;
+    comptroller._setCollateralFactor(bWETH.address, utils.parseUnits('0.8', 18));
+    bBAL = market.bBAL;
+    comptroller._setCollateralFactor(bWETH.address, utils.parseUnits('0.8', 18));
+    bUSDC = market.bUSDC;
+    comptroller._setCollateralFactor(bWETH.address, utils.parseUnits('0.8', 18));
+    bWBTC = market.bWBTC;
+    comptroller._setCollateralFactor(bWETH.address, utils.parseUnits('0.8', 18));
+    bOHM = market.bOHM;
+    comptroller._setCollateralFactor(bWETH.address, utils.parseUnits('0.8', 18));
+
+    await faucetToken(ADDRESS.WETH, utils.parseUnits('100', 18), admin);
+  });
+
+  beforeEach(async () => {
+    // Transfer WETH to alice
+    const weth = await ethers.getContractAt('IERC20', ADDRESS.WETH, admin);
+    await weth.connect(admin).transfer(alice.address, utils.parseUnits('1', 18));
+
+    // Lend WETH to the market
+    await weth.connect(alice).approve(bWETH.address, utils.parseUnits('1', 18));
+    await bWETH.connect(alice).mint(utils.parseUnits('1', 18));
+    await comptroller.connect(alice).enterMarkets([bWETH.address]);
+    await comptroller._setCollateralFactor(bWETH.address, utils.parseUnits('0.8', 18));
+  });
+
+  it('successfully borrow from a market | 18 decimals', async () => {
+    const seedAmount = utils.parseUnits('10000', 18);
+    const underlying = await ethers.getContractAt('ERC20', await bBAL.underlying(), admin);
+    await seedMarket(bBAL, underlying, admin, seedAmount);
+
+    // Borrow bBAL from the market
+    const borrowAmount = utils.parseUnits('4', 18);
+    await bBAL.connect(alice).borrow(borrowAmount);
```

Comment:
I believe this will also fail because of #172, but it can be fixed by simply giving `alice` a credit limit.